source/dem/insertion.cc

```diff
@@ -21,6 +21,28 @@
 
 #include <sstream>
 
+template <int dim>
+Insertion<dim>::Insertion(const DEMSolverParameters<dim> &dem_parameters)
+{
+  if (dem_parameters.lagrangian_physical_properties.size_distribution_type ==
+      Parameters::Lagrangian::LagrangianPhysicalProperties::
+        size_distribution_type::uniform)
+    {
+      distribution_object = std::make_shared<NormalDistribution>(
+        dem_parameters.lagrangian_physical_properties.particle_average_diameter,
+        dem_parameters.lagrangian_physical_properties.particle_size_std);
+    }
+  else if (dem_parameters.lagrangian_physical_properties
+             .size_distribution_type ==
+           Parameters::Lagrangian::LagrangianPhysicalProperties::
+             size_distribution_type::normal)
+    {
+      distribution_object = std::make_shared<NormalDistribution>(
+        dem_parameters.lagrangian_physical_properties.particle_average_diameter,
+        dem_parameters.lagrangian_physical_properties.particle_size_std);
+    }
+}
```

Comment:
I don't understand this.... why are we doing the same if the distribution is uniform and normal? Or is it a WIP?

Reply:
Good question. An uniform distribution is just a normal distribution with a standard deviation equal to zero. This is how the code was working previously. (If you check in the parsing function,dem_parameters.lagrangian_physical_properties.particle_size_std  is set to zero when the distribution type is uniform)

We could add a new class called UniformDistribution with its own constructor, but I think that this would go in a other PR.

Reply:
Somehow I think it would be cleaner but I am not sure... maybe if we do not want to create a new class, then we can use the same one but I would add a comment within this if so it would be clear for anybody reading the code

Reply:
But then shouldnt at least the standard deviation be fixed to zero?
Maybe we should deprecate uniform completely and just ask the user to write normal with no standard deviation. Seems like a very weird option

Reply:
When the distribution is set to unifom, the standard deviation is fixed to zero in the parsing function. The NormalDistribution constructor takes two std::unordered_map as inputs to support the uses of many types of particles, that is why it is set to zero there and not here. (Otherwise, I would have to create a std::unordered_map here, which would have been weird)

I could deprecated the uniform size distribution... not sure if it is a good thing, especially when you consider new types of size distribution that are going to be added in the following weeks.

Reply:
Well I think it's better to hardcode the parameter in the constructor than in the parser of the parameters
In all cases, uniform should always always have a std-dev of 0, no point in passing a st-dev assuming it has been set to zero previously

Reply:
I like more the idea of having a separate class for the uniform distribution without a standard deviation parameter. This way when other distributions would be implemented, it will be clear that another class need to be implemented and nothing would get mixed.

Reply:
I will do separate class, this will solve the problem.